src/subdomains/monitoring/observers/payment.observer.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import { Interval } from '@nestjs/schedule';
+import { RewardRepository } from 'src/subdomains/staking/application/repositories/reward.repository';
+import { StakingStrategy } from 'src/subdomains/staking/domain/enums';
+import { getCustomRepository, In, IsNull, Not } from 'typeorm';
+import { MonitoringService } from '../application/services/monitoring.service';
+import { MetricObserver } from '../metric.observer';
+
+interface PaymentData {
+  lastOutputDates: LastOutputDates;
+}
+
+interface LastOutputDates {
+  yieldMachineReward: Date;
+  stakingReward: Date;
+}
+
+@Injectable()
+export class PaymentObserver extends MetricObserver<PaymentData> {
```

Comment:
Is payment the right name? Maybe rather transaction? Or move everything to the staking-combined observer?